src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -317,33 +294,48 @@ private int instances(AppDeploymentRequest request) {
 	}
 
 	private Mono<Void> pushApplication(String deploymentId, AppDeploymentRequest request) {
-		return requestPushApplication(PushApplicationRequest.builder()
-			.application(getApplication(request)) // Only one of the two is non-null
+		ApplicationManifest.Builder manifest = ApplicationManifest.builder()
+			.path(getApplication(request)) // Only one of the two is non-null
 			.dockerImage(getDockerImage(request)) // Only one of the two is non-null
 			.buildpack(buildpack(request))
-			.diskQuota(diskQuota(request))
-			.domain(domain(request))
+			.disk(diskQuota(request))
+			.environmentVariables(getEnvironmentVariables(deploymentId, request))
 			.healthCheckType(healthCheck(request))
-			.host(host(request))
 			.instances(instances(request))
 			.memory(memory(request))
 			.name(deploymentId)
 			.noRoute(toggleNoRoute(request))
-			.noStart(true)
-			.routePath(routePath(request))
-			.build())
+			.services(servicesToBind(request));
+
+		String domain = domain(request);
+		String host = host(request);
+		String routePath = routePath(request);
+
+		if (routePath != null) {
+			manifest
+				.route(Route.builder()
+					.route(String.format("%s.%s%s", host, domain, routePath))
```

Comment:
That's what I realized afterwards indeed. So your checks in the following lines are not needed either, correct?